src/coreclr/vm/stringliteralmap.cpp

```diff
@@ -631,10 +645,13 @@ void StringLiteralEntry::DeleteEntry (StringLiteralEntry *pEntry)
     }
     CONTRACTL_END;
 
-    _ASSERTE (VolatileLoad(&pEntry->m_dwRefCount) == 0);
+    _ASSERTE (pEntry->GetRefCount() == 0);
 
 #ifdef _DEBUG
-    memset (&pEntry->m_pStringObj, 0xc, sizeof(pEntry->m_pStringObj));
+    if (!pEntry->IsStringFrozen())
```

Comment:
Can `DeleteEntry` be ever called on frozen strings? The entries for frozen strings should not be ever deleted.

Reply:
Added an assert